### PR TITLE
Fixed the infantry death animations

### PR DESCRIPTION
**Pull request by Mailaender**
Followup of https://github.com/OpenRA/OpenRA/pull/11391.

### CODE REVIEW

File: rules/defaults.yaml [hunk 1]
```diff
@@ -278,13 +278,15 @@
...
 			ExplosionDeath: 1
 			BulletDeath: 2
+			ElectroDeath: 6
 	WithDeathAnimation@effect:
 		DeathSequencePalette: effect
 		DeathPaletteIsPlayerPalette: False
 		DeathTypes:
 			RadiationDeath: 4
-			FlameDeath: 5
-			ElectroDeath: 6
```
Comment (GraionDilach) [inline rules/defaults.yaml]: Do not change this, this is valid. electro uses anim.pal and not unittem.pal.

Comment (GraionDilach) [inline rules/defaults.yaml]: Ah, we have an SHP override in our bits folder which is indeed converted to unittem - removing that file should be the fix.

Comment (GraionDilach) [inline rules/defaults.yaml]: This same applies to flameguy as well.

Comment (Mailaender) [inline rules/defaults.yaml]: Removed the SHP workarounds. I didn't know they existed.

File: rules/misc.yaml [hunk 2]
```diff
@@ -105,3 +105,34 @@ CAMERA:
...
+		HP: 160
+	SelfHealing:
+		Step: -10
+		HealIfBelow: 101
+	ScaredyCat:
+	WithDeathAnimation:
+		FallbackSequence: die
+		UseDeathTypeSuffix: false
+		DeathSequencePalette: effect
+		DeathPaletteIsPlayerPalette: False
```
Comment (abcdefg30) [inline rules/misc.yaml]: Minor nit: Please use a lowercase `false` as we/you did in the rest of the file.

**Comment by abcdefg30** [comment]
Looks good to me otherwise. 👍 / ✅
Thanks!

File: rules/misc.yaml (hunk 2 above)
Comment (Mailaender) [inline rules/misc.yaml]: Fixed.

**Pull request merged**